### PR TITLE
Various UI improvments and fixes

### PR DESCRIPTION
**Pull request by EttyKitty**
### Things that are done:
1. Fixed a rare crash related to the Damage Resistance tooltip.
2. Added sources of the HP mods to its tooltip.
[Preview](https://github.com/OH296/ChapterMaster/assets/20323032/fc37da87-3f57-4ed7-86b6-a6f27c9d8e00)
3. Improved description of Ultramarines starting artifacts.
[Preview](https://github.com/OH296/ChapterMaster/assets/20323032/3588d57d-d2d0-4743-9576-728b7835118f)
[Preview](https://github.com/OH296/ChapterMaster/assets/20323032/493b332e-01d0-4ede-ac5c-0f1157200753)
4. Added a new attribute for items - special_properties or Properties.
This thing is basically just an array of things that were previously put into special_description. At the tooltip stage, armor piercing property or low penetration also gets added to this array and are displayed under Properties in the tooltip.
[Preview](https://github.com/OH296/ChapterMaster/assets/20323032/ce4ffa0d-9b7d-4735-bf03-e9915cc03831)
5. Adjusted artifact description box (I think it was something different in old versions). See Ultramarines artifacts description for screenshots.
6. Added forge points to the top bar.
[Preview](https://github.com/OH296/ChapterMaster/assets/20323032/ea16113a-ee3b-405c-ac25-0290dcdc7865)
7. Improved tooltips for some of the top bar stuff.
8. Made some alignment, color and text adjustments to the forge screen. Removed the icon, because you can always see points production in the top bar.
[Preview](https://github.com/OH296/ChapterMaster/assets/20323032/a678d6ea-f6a9-4f5a-ae36-034b88a929e5)
9. Some small adjustments to the shop screen. Proper if checks for displayed values/headers and icon scaling.
10. [Display company captain's name on the Chapter Management screen](https://github.com/OH296/ChapterMaster/commit/aa80dcdd539bfb610715c6f4f00b482b2fb3d97f)
[Preview](https://github.com/OH296/ChapterMaster/assets/20323032/5930ea37-ff91-4b96-aa29-a3b8577d5ef8)
10. Small touches to various abbreviations and descriptions. Typo fixes, case fixes and similar minor stuff.

### CODE REVIEW

File: scripts/scr_initialize_custom/scr_initialize_custom.gml [hunk 1]
```diff
@@ -1359,7 +1360,7 @@ function scr_initialize_custom() {
 			chapter_master_equip.armour = last_artifact;
 			arti = obj_ini.artifact_struct[last_artifact];
 			arti.name = "Armour of Antilochus";
-			arti.custom_description="a masterwork suit of the standard Indomitus pattern Terminator Armour"
+			arti.custom_description="A masterwork suit of the standard Indomitus pattern Terminator Armour. It incorporates a Teleport Homer, allowing Terminator squads of the veteran First Company to deploy next to their Chapter Master's side."
```
Comment (OH296) [inline scripts/scr_initialize_custom/scr_initialize_custom.gml]:
```suggestion
			arti.custom_description="A masterwork suit of the standard Indomitus pattern Terminator Armour. It incorporates a Teleport Homer, allowing Terminator squads of the veteran First Company to deploy next to their Chapter Master's side.";
```

File: scripts/scr_initialize_custom/scr_initialize_custom.gml [hunk 2]
```diff
@@ -1349,6 +1349,7 @@ function scr_initialize_custom() {
 			chapter_master.add_trait("tyrannic_vet");
 			arti = obj_ini.artifact_struct[last_artifact];
 			arti.name = "Gauntlets of Ultramar";
+			arti.custom_description="Two mighty Power Fists with incorporated Bolters that were reclaimed from a fallen Chaos champion, slain during the Gamalia Reclusiam Massacre by the Primarch of the Ultramarines, Roboute Guilliman himself"
```
Comment (OH296) [inline scripts/scr_initialize_custom/scr_initialize_custom.gml]:
```suggestion
			arti.custom_description="Two mighty Power Fists with incorporated Bolters that were reclaimed from a fallen Chaos champion, slain during the Gamalia Reclusiam Massacre by the Primarch of the Ultramarines, Roboute Guilliman himself";
```

File: scripts/scr_management/scr_management.gml [hunk 3]
```diff
@@ -337,7 +337,7 @@ function scr_management(argument0) {
 			}
         
 			// Indexing the names to nam array
-	        nam[1]=obj_ini.role[100][5];
```
Comment (OH296) [inline scripts/scr_management/scr_management.gml:340]: if taking this approach for ease fo use may as well use a global enum? probably easier than having the same name array repeated all over? probably fine in this instance but if for any reason this needs repeating else where revert to enums and a global fix.

**Review by OH296 — APPROVED** [review]
Basically all fine will merge later

**Comment by EttyKitty** [comment]
Found a bug. Need to fix it first.

**Pull request merged**